web/src/hooks/transaction/useSubmitTransaction.ts

```diff
@@ -59,104 +43,89 @@ const useSubmitTransaction = (
+    if (!accountClient || !account) {
+      console.log('accountClient', accountClient);
+      console.log('account', account);
+      setError('No account connected');
+      return;
+    }
 
-  const error = rawError
-    ? processViemContractError(rawError, (errorName) => {
-        if (!options?.customErrorsMap || !(errorName in options.customErrorsMap))
-          return `Contract error: ${errorName}`;
-        return options.customErrorsMap[errorName];
+    setIsLoading(true);
+    setError(undefined);
+
+    const transactions = Array.isArray(contractCallConfig.contracts)
+      ? contractCallConfig.contracts
+      : [contractCallConfig];
+
+    accountClient
+      .sendTransactions({
+        transactions: transactions.map((contract: any) => ({
+          to: contract.address,
+          data: encodeFunctionData({
+            abi: contract.abi,
+            functionName: contract.functionName,
+            args: contract.args,
+          }),
+          value: BigInt(contract.value ?? 0),
+        })),
+        account: account,
       })
-    : undefined;
-
-  const isError = isBatchWriteContractError || isWaitForTransactionError;
-  const { onSuccess, onError } = options ?? {};
+      .then((txHash) => {
+        setTransactionHash(txHash);
+        onSuccessCalledRef.current = false;
+        options?.onSent?.();
+      })
+      .catch((err: any) => {
+        const errorMessage = processViemContractError(err, (errorName) => {
+          if (!options?.customErrorsMap || !(errorName in options.customErrorsMap))
+            return `Contract error: ${errorName}`;
+          return options.customErrorsMap[errorName];
+        });
+        setError(errorMessage ?? 'Unknown error occurred');
+        options?.onError?.(errorMessage ?? 'Unknown error occurred', err);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  };
 
   useEffect(() => {
-    if (!transactionReceipt && !isSuccess && !isError) return;
-
-    if (transactionReceipt && isSuccess) {
-      if (typeof onSuccess === 'function') {
-        const events = getEvents(contractCallConfig, transactionReceipt);
-
-        onSuccess(
-          transactionReceipt as TransactionReceipt,
-          events as unknown as DecodeEventLogReturnType[],
-        );
-        setIsLoading(false);
-      }
+    if (transactionReceipt && options?.onSuccess && !onSuccessCalledRef.current) {
+      const events = getEvents(contractCallConfig, transactionReceipt);
+      options.onSuccess(transactionReceipt, events as DecodeEventLogReturnType[]);
+      onSuccessCalledRef.current = true; // Mark onSuccess as called for this transaction
     }
+  }, [transactionReceipt, options, contractCallConfig]);
 
-    if (error) {
-      onError?.(error, rawError);
-      setIsLoading(false);
-      resetBatchWriteContract();
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [transactionReceipt, isSuccess, isError]);
+  useEffect(() => {
```

Comment:
In my understanding, we won't need this part. Cause if the transaction failed, this hook will be triggered and create a new mount anyway.